.github/workflows/macos-nightly.yml

```diff
@@ -0,0 +1,40 @@
+name: Celix MacOS Nightly
+
+on:
+  schedule:
+    - cron:  '0 0 * * *'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macOS-latest]
+        compiler: [clang]
+    timeout-minutes: 120
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@master
+    - name: Install dependencies
+      run: |
+        brew update
+        brew install lcov zeromq czmq openssl cpputest
+        brew unlink openssl && brew link openssl --force
+    - name: Build
+      env:
+        CC: ${{ matrix.compiler }}
+        BUILD_OPTIONS: |
+          -DENABLE_TESTING=ON
+          -DENABLE_ADDRESS_SANITIZER=ON
+      run: |
+        mkdir build install
+        cd build
+        cmake -DCMAKE_BUILD_TYPE=Release ${BUILD_OPTIONS} -DCMAKE_INSTALL_PREFIX=../install ..
```

Comment:
I prefer the default build to be RelWithDebInfo. Especially if that is going to be the build type without postfixes for bundles.

Reply:
I updated the nightly to use RelWithDefInfo instead of Release.

But running test with RelWithDefInfo or Release on my Ubuntu PC fails on the rsa dfi tests.

Reply:
The fails are due to memory leaks and not on test dfi, but rsa dfi. Because of that I think is better for now to disable the rsa dfi test on Release/RelWithDebInfo. 
Than we the "normal" release compiler settings.

Reply:
Are you now also referring to the disabled optimization? Because that one does fail due to tetst_dfi.
Also, specifically due to some strange behaviour in the ffi lib itself. It might be memory leaks...
Anyway, whatever it is, since test_dfi fails with O1 (or 2 or 3) due to an external lib, I don't think it is wise to disable a test and build with optimization. Since the actual problem of that case is not in the test..

Reply:
On my system only rsa dfi fails when using -O2

Reply:
What is the ffi version that you are using? 

I have, on MacOS 10.14.6:
* MacOS ffi, with which Celix doesn't compile at all
* Brew version 3.3 (which is the latest release), on which test_dfi always fails when using optimization
And on Ubuntu 20.04
* 3.3 on which it also fails in the same way

I think @Oipo also tested it, what version of ffi are you using?

Reply:
Specifically compiling this branch with Debug mode, running libs/dfi/gtest/test_dfi gives me the following backtrace:
```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7b6a899 in __GI_abort () at abort.c:79
#2  0x00007ffff7b6a769 in __assert_fail_base (fmt=0x7ffff7cfefb8 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=0x5555555fbaf5 "desc != NULL", file=0x5555555fba50 "/home/oipo-unencrypted/Programming/celix-apache/libs/dfi/gtest/src/dyn_interface_tests.cpp", line=73, function=<optimized out>) at assert.c:92
#3  0x00007ffff7b7c006 in __GI___assert_fail (assertion=0x5555555fbaf5 "desc != NULL", file=0x5555555fba50 "/home/oipo-unencrypted/Programming/celix-apache/libs/dfi/gtest/src/dyn_interface_tests.cpp", line=73, function=0x5555555fbae8 "void test1()") at assert.c:101
#4  0x00005555555817de in test1 () at /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/gtest/src/dyn_interface_tests.cpp:73
#5  0x0000555555583269 in DynInterfaceTests_test1_Test::TestBody (this=0x55555566ac00) at /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/gtest/src/dyn_interface_tests.cpp:205
#6  0x00005555555e1cc5 in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void> (object=0x55555566ac00, method=&virtual testing::Test::TestBody(), location=0x555555606243 "the test body")
    at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2433
#7  0x00005555555da58d in testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void> (object=0x55555566ac00, method=&virtual testing::Test::TestBody(), location=0x555555606243 "the test body") at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2469
#8  0x00005555555b4ed2 in testing::Test::Run (this=0x55555566ac00) at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2508
#9  0x00005555555b58bd in testing::TestInfo::Run (this=0x55555565a270) at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2684
#10 0x00005555555b5ff8 in testing::TestSuite::Run (this=0x55555565a440) at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2816
#11 0x00005555555c20b4 in testing::internal::UnitTestImpl::RunAllTests (this=0x5555556521b0) at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:5338
#12 0x00005555555e31ee in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (object=0x5555556521b0, method=(bool (testing::internal::UnitTestImpl::*)(class testing::internal::UnitTestImpl * const)) 0x5555555c1cac <testing::internal::UnitTestImpl::RunAllTests()>, 
    location=0x555555606c80 "auxiliary test code (environments or event listeners)") at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2433
#13 0x00005555555db7cb in testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (object=0x5555556521b0, method=(bool (testing::internal::UnitTestImpl::*)(class testing::internal::UnitTestImpl * const)) 0x5555555c1cac <testing::internal::UnitTestImpl::RunAllTests()>, 
    location=0x555555606c80 "auxiliary test code (environments or event listeners)") at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2469
#14 0x00005555555c08f4 in testing::UnitTest::Run (this=0x55555563f940 <testing::UnitTest::GetInstance()::instance>) at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:4925
#15 0x00005555555f597f in RUN_ALL_TESTS () at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/include/gtest/gtest.h:2473
#16 0x00005555555f5901 in main (argc=1, argv=0x7fffffffdbb8) at /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest_main.cc:45
```

Compiling with debug and -O2 and running bundles/remote_services/remote_service_admin_dfi/gtest/test_rsa_dfi gives me the following failed tests:
```
[==========] Running 11 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 4 tests from RsaDfiTests
[ RUN      ] RsaDfiTests.InfoTest
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_tests.cc:40: Failure
Value of: fwProperties != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiTests.InfoTest (0 ms)
[ RUN      ] RsaDfiTests.ExportService
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_tests.cc:40: Failure
Value of: fwProperties != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiTests.ExportService (0 ms)
[ RUN      ] RsaDfiTests.ImportService
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_tests.cc:40: Failure
Value of: fwProperties != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiTests.ImportService (0 ms)
[ RUN      ] RsaDfiTests.TestBundles
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_tests.cc:40: Failure
Value of: fwProperties != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiTests.TestBundles (0 ms)
[----------] 4 tests from RsaDfiTests (0 ms total)

[----------] 7 tests from RsaDfiClientServerTests
[ RUN      ] RsaDfiClientServerTests.TestRemoteCalculator
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_client_server_tests.cc:47: Failure
Value of: serverProps != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiClientServerTests.TestRemoteCalculator (0 ms)
[ RUN      ] RsaDfiClientServerTests.TestRemoteComplex
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_client_server_tests.cc:47: Failure
Value of: serverProps != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiClientServerTests.TestRemoteComplex (0 ms)
[ RUN      ] RsaDfiClientServerTests.TestRemoteNumbers
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_client_server_tests.cc:47: Failure
Value of: serverProps != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiClientServerTests.TestRemoteNumbers (0 ms)
[ RUN      ] RsaDfiClientServerTests.TestRemoteString
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_client_server_tests.cc:47: Failure
Value of: serverProps != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiClientServerTests.TestRemoteString (0 ms)
[ RUN      ] RsaDfiClientServerTests.TestRemoteConstString
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_client_server_tests.cc:47: Failure
Value of: serverProps != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiClientServerTests.TestRemoteConstString (0 ms)
[ RUN      ] RsaDfiClientServerTests.TestRemoteEnum
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_client_server_tests.cc:47: Failure
Value of: serverProps != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiClientServerTests.TestRemoteEnum (0 ms)
[ RUN      ] RsaDfiClientServerTests.TestRemoteAction
/home/oipo-unencrypted/Programming/celix-apache/bundles/remote_services/remote_service_admin_dfi/gtest/src/rsa_client_server_tests.cc:47: Failure
Value of: serverProps != NULL
  Actual: false
Expected: true
[  FAILED  ] RsaDfiClientServerTests.TestRemoteAction (0 ms)
[----------] 7 tests from RsaDfiClientServerTests (0 ms total)

[----------] Global test environment tear-down
[==========] 11 tests from 2 test suites ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 11 tests, listed below:
[  FAILED  ] RsaDfiTests.InfoTest
[  FAILED  ] RsaDfiTests.ExportService
[  FAILED  ] RsaDfiTests.ImportService
[  FAILED  ] RsaDfiTests.TestBundles
[  FAILED  ] RsaDfiClientServerTests.TestRemoteCalculator
[  FAILED  ] RsaDfiClientServerTests.TestRemoteComplex
[  FAILED  ] RsaDfiClientServerTests.TestRemoteNumbers
[  FAILED  ] RsaDfiClientServerTests.TestRemoteString
[  FAILED  ] RsaDfiClientServerTests.TestRemoteConstString
[  FAILED  ] RsaDfiClientServerTests.TestRemoteEnum
[  FAILED  ] RsaDfiClientServerTests.TestRemoteAction
```

Compiling in release mode with -O2 and running bundles/remote_services/remote_service_admin_dfi/gtest/test_rsa_dfi gives me a segfault:
```
#0  0x00007ffff7b49fdb in ffi_call () from /lib/x86_64-linux-gnu/libffi.so.6
#1  0x00007ffff7fb7b51 in dynFunction_call () from /home/oipo-unencrypted/Programming/celix-apache/build/libs/dfi/libcelix_dfi.so.1
#2  0x0000555555581873 in DynFunctionTests_DynFuncTest3_Test::TestBody() ()
#3  0x00005555555d9de1 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) ()
#4  0x00005555555cd3ea in testing::Test::Run() [clone .part.0] ()
#5  0x00005555555cd882 in testing::TestInfo::Run() [clone .part.0] ()
#6  0x00005555555cda8e in testing::TestSuite::Run() [clone .part.0] ()
#7  0x00005555555ceb4d in testing::internal::UnitTestImpl::RunAllTests() ()
#8  0x00005555555cf0b8 in testing::UnitTest::Run() ()
#9  0x0000555555577b34 in main ()
```

Reply:
This is on Ubuntu 19.10 with the following ffi installed:
```
libffi-dev/eoan,now 3.2.1-9 amd64 [installed]
libffi6/eoan,now 3.2.1-9 amd64 [installed,automatic]
libffi6/eoan,now 3.2.1-9 i386 [installed,automatic]
```

Reply:
Strangely, the libs/dfi/gtest/test_dfi executable seems to not be included in the tests that are run?

Reply:
I am using Ubuntu 18 with 
libffi-dev/bionic,now 3.2.1-8 amd64 [installed]

And only rsa_dfi test fail on Release with -O2 with a mem leak. For Debug everything is green.

Reply:
This just doesn't feel good. I don't really like the idea of using optimization with the unexpected behaviour for different parts on different systems.

Can we, for now, just use no optimization. That way we are sure tests do work, and ffi is not a problem.

Just curious, as @Oipo indicates, he doesn't see test_dfi, are you sure it does run on your system? Or is it just not being executed?

Reply:
Disregard not seeing test_dfi. I must've missed it, it runs in the make test command.

Also, the test_rsa_dfi errors were due to mis-compilations. If you make clean and make in the `bundles/remote_services/remote_service_admin_dfi/gtest` directory, the test fails. But compiling it normally from the top-level directory, it works.

Reply:
There's something weird going on with the dynfunction tests:
```
[ RUN      ] DynFunctionTests.DynFuncTest2
=================================================================
==64079==ERROR: AddressSanitizer: dynamic-stack-buffer-overflow on address 0x7fff76e507b0 at pc 0x7dd750145945 bp 0x7fff76e50740 sp 0x7fff76e50730
WRITE of size 8 at 0x7fff76e507b0 thread T0
    #0 0x7dd750145944 in ffi_call_int (/home/oipo-unencrypted/Programming/ffi/lib/libffi.so.7+0x56944)
    #1 0x7dd7511815e6 in dynFunction_call /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/src/dyn_function.c:240
    #2 0x5591d28e4e5f in func_test2 /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/gtest/src/dyn_function_tests.cpp:113
    #3 0x5591d28ea093 in DynFunctionTests_DynFuncTest2_Test::TestBody() /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/gtest/src/dyn_function_tests.cpp:124
    #4 0x5591d2b56597 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2433
    #5 0x5591d2b3d868 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2469
    #6 0x5591d2ac8209 in testing::Test::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2508
    #7 0x5591d2acac6d in testing::TestInfo::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2684
    #8 0x5591d2acda4b in testing::TestSuite::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2816
    #9 0x5591d2af5a79 in testing::internal::UnitTestImpl::RunAllTests() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:5338
    #10 0x5591d2b5a8c4 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2433
    #11 0x5591d2b41b19 in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2469
    #12 0x5591d2aebe1e in testing::UnitTest::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:4925
    #13 0x5591d2b89ea8 in RUN_ALL_TESTS() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/include/gtest/gtest.h:2473
    #14 0x5591d2b89d54 in main /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest_main.cc:45
    #15 0x7dd75031d1e2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x271e2)
    #16 0x5591d2878a3d in _start (/home/oipo-unencrypted/Programming/celix-apache/build/libs/dfi/gtest/test_dfi+0x526a3d)

Address 0x7fff76e507b0 is located in stack of thread T0
SUMMARY: AddressSanitizer: dynamic-stack-buffer-overflow (/home/oipo-unencrypted/Programming/ffi/lib/libffi.so.7+0x56944) in ffi_call_int
Shadow bytes around the buggy address:
  0x10006edc20a0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10006edc20b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10006edc20c0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10006edc20d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10006edc20e0: 00 00 00 00 00 00 00 00 00 00 00 00 ca ca ca ca
=>0x10006edc20f0: 00 00 00 00 ca ca[ca]ca 00 00 00 00 00 00 00 00
  0x10006edc2100: 00 00 00 00 00 00 00 00 00 00 00 00 cb cb cb cb
  0x10006edc2110: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10006edc2120: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 f1 f1
  0x10006edc2130: f1 f1 04 f2 04 f2 00 00 f3 f3 00 00 00 00 00 00
  0x10006edc2140: 00 00 00 00 00 00 00 00 00 00 00 00 f1 f1 f1 f1
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==64079==ABORTING
```

```
[ RUN      ] JsonAvprRpcTests.prep
=================================================================
==64243==ERROR: AddressSanitizer: dynamic-stack-buffer-overflow on address 0x7ffe09a5385c at pc 0x7a61dca6f480 bp 0x7ffe09a53810 sp 0x7ffe09a52fb8
READ of size 1 at 0x7ffe09a5385c thread T0
    #0 0x7a61dca6f47f  (/lib/x86_64-linux-gnu/libasan.so.5+0x9b47f)
    #1 0x7a61db8bfdab  (/lib/x86_64-linux-gnu/libjansson.so.4+0x7dab)
    #2 0x7a61db8bfdf7  (/lib/x86_64-linux-gnu/libjansson.so.4+0x7df7)
    #3 0x7a61db8bc63a  (/lib/x86_64-linux-gnu/libjansson.so.4+0x463a)
    #4 0x7a61db8bc928  (/lib/x86_64-linux-gnu/libjansson.so.4+0x4928)
    #5 0x7a61db8bd05f  (/lib/x86_64-linux-gnu/libjansson.so.4+0x505f)
    #6 0x7a61db8bd255  (/lib/x86_64-linux-gnu/libjansson.so.4+0x5255)
    #7 0x7a61db8bd0e6  (/lib/x86_64-linux-gnu/libjansson.so.4+0x50e6)
    #8 0x7a61db8bd255  (/lib/x86_64-linux-gnu/libjansson.so.4+0x5255)
    #9 0x7a61db8bd0e6  (/lib/x86_64-linux-gnu/libjansson.so.4+0x50e6)
    #10 0x7a61db8bd395  (/lib/x86_64-linux-gnu/libjansson.so.4+0x5395)
    #11 0x7a61db8bd756 in json_loadf (/lib/x86_64-linux-gnu/libjansson.so.4+0x5756)
    #12 0x7a61dc95c031 in dynFunction_parseAvpr /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/src/dyn_avpr_function.c:57
    #13 0x7a61dc95c2a5 in dynFunction_parseAvprWithStr /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/src/dyn_avpr_function.c:76
    #14 0x5c2ddfefcb6b in JsonAvprRpcTests_prep_Test::TestBody() /home/oipo-unencrypted/Programming/celix-apache/libs/dfi/gtest/src/json_rpc_avpr_tests.cpp:173
    #15 0x5c2ddfffa1db in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2433
    #16 0x5c2ddffe14ac in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2469
    #17 0x5c2ddff6be4d in testing::Test::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2508
    #18 0x5c2ddff6e8b1 in testing::TestInfo::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2684
    #19 0x5c2ddff7168f in testing::TestSuite::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2816
    #20 0x5c2ddff996bd in testing::internal::UnitTestImpl::RunAllTests() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:5338
    #21 0x5c2ddfffe508 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2433
    #22 0x5c2ddffe575d in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:2469
    #23 0x5c2ddff8fa62 in testing::UnitTest::Run() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest.cc:4925
    #24 0x5c2de002daec in RUN_ALL_TESTS() /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/include/gtest/gtest.h:2473
    #25 0x5c2de002d998 in main /home/oipo-unencrypted/Programming/celix-apache/build/_deps/googletest-src/googletest/src/gtest_main.cc:45
    #26 0x7a61dbaf51e2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x271e2)
    #27 0x5c2ddfd1ca3d in _start (/home/oipo-unencrypted/Programming/celix-apache/build/libs/dfi/gtest/test_dfi+0x526a3d)

Address 0x7ffe09a5385c is located in stack of thread T0
SUMMARY: AddressSanitizer: dynamic-stack-buffer-overflow (/lib/x86_64-linux-gnu/libasan.so.5+0x9b47f) 
Shadow bytes around the buggy address:
  0x1000413426b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1000413426c0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1000413426d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1000413426e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x1000413426f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x100041342700: 00 00 00 00 00 00 00 00 ca ca ca[ca]00 00 00 00
  0x100041342710: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100041342720: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100041342730: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100041342740: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100041342750: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==64243==ABORTING
```